src/community/features-autopopulate/features-autopopulate-core/src/main/java/org/geoserver/autopopulate/AutopopulateTemplate.java

```diff
@@ -0,0 +1,118 @@
+/* (c) 2024 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.autopopulate;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Properties;
+import java.util.logging.Logger;
+import org.geotools.api.filter.expression.Expression;
+import org.geotools.filter.text.cql2.CQLException;
+import org.geotools.filter.text.ecql.ECQL;
+import org.geotools.util.logging.Logging;
+
+/**
+ * AutopopulateTemplate class is used to load the properties from the file and store them in a map
+ * for further use.
+ *
+ * <p>It also provides the methods to get the properties and set the properties.
+ *
+ * @author Alessio Fabiani, GeoSolutions SRL, alessio.fabiani@geosolutionsgroup.com
+ */
+public class AutopopulateTemplate {
+    /** logger */
+    private static final Logger log = Logging.getLogger(AutopopulateTransactionCallback.class);
```

Comment:
By common naming convention (usage points need to be updated as well):

```suggestion
    private static final Logger LOGGER = Logging.getLogger(AutopopulateTransactionCallback.class);
```